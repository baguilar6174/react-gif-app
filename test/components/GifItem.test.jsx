import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe('Tests on <GifItem /> component', () => {

  const title = "Master Chief Halo GIF by Xbox";
  const url = "https://media2.giphy.com/media/b4ab9TBk9Ornvrt9W8/giphy-downsized-medium.gif?cid=8c7595b4vldb5u0mu6eldps3w7w0hcatlqsudgtzz74uuwox&rid=giphy-downsized-medium.gif&ct=g";

  test('should be match with the snapshot', () => {
    const { container } = render(
      <GifItem title={title} url={url} />
    );
    expect(container).toMatchSnapshot();
  });

  test('should be show rhe image with url and alt', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);

    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);

  });

  test('should be show the title', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

});