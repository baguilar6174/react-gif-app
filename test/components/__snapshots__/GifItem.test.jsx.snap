// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> component should be match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Master Chief Halo GIF by Xbox"
      src="https://media2.giphy.com/media/b4ab9TBk9Ornvrt9W8/giphy-downsized-medium.gif?cid=8c7595b4vldb5u0mu6eldps3w7w0hcatlqsudgtzz74uuwox&rid=giphy-downsized-medium.gif&ct=g"
    />
    <p>
      Master Chief Halo GIF by Xbox
    </p>
  </div>
</div>
`;
